lnwallet/btcwallet/btcwallet.go

```diff
@@ -353,17 +353,36 @@ func (b *BtcWallet) ListUnspentWitness(minConfs int32) ([]*lnwallet.Utxo, error)
 			return nil, err
 		}
 
-		// TODO(roasbeef): this assumes all p2sh outputs returned by
-		// the wallet are nested p2sh...
-		if txscript.IsPayToWitnessPubKeyHash(pkScript) ||
-			txscript.IsPayToScriptHash(pkScript) {
+		var addressType lnwallet.AddressType
+		var redeemScript []byte
+
+		if txscript.IsPayToWitnessPubKeyHash(pkScript) {
+			addressType = lnwallet.WitnessPubKey
+		} else if txscript.IsPayToScriptHash(pkScript) {
+			// Even if it is not a native witness output, it could be a witness
+			// output nested in P2SH.
+			redeemScript, err = hex.DecodeString(output.RedeemScript)
+			if err != nil {
+				return nil, err
+			}
+			if txscript.IsPayToWitnessPubKeyHash(redeemScript) {
```

Comment:
This doesn't quite work (atm) as the `RedeemScript` is never actually populated within the `ListUnspent` implementation in `btcwallet`. 

Until this is resolved within `btcwallet`, we can fallback to the prior behavior of assuming all p2sh outputs are themselves nested p2sh outputs.